genienlp/util.py

```diff
@@ -102,6 +104,70 @@ def backward(self, s: str, occurrence: str):
             
         return s
 
+
+class ConfidenceOutput:
+    """
+    Contains all necessary features that are useful for calculating confidence of a single generated output
+    """
```

Comment:
I find the naming of this class (and the flag "output_confidence") to be quite confusing. It doesn't contain the confidence, it contains the features to compute the confidence. So maybe "ConfidenceFeatures" ? I know it would be a bit confusing with the low-level features in confidence.py, but at least the confusion is contained in one file instead of being across many files and API boundaries.